hedera-node/src/main/java/com/hedera/services/ledger/interceptors/StakeAwareAccountsCommitsInterceptor.java

```diff
@@ -0,0 +1,317 @@
+		// reward; if staking is not active, this will return Long.MIN_VALUE so no account is eligible
+		final var latestEligibleStart = stakePeriodManager.latestRewardableStakePeriodStart();
+
+		// boolean to track if the account has been rewarded already with one of the pending changes
+		hasBeenRewarded = new HashSet<>();
+
+		for (int i = 0, n = pendingChanges.size(); i < n; i++) {
+			final var account = pendingChanges.entity(i);
+			final var changes = pendingChanges.changes(i);
+			final var accountNum = pendingChanges.id(i).getAccountNum();
+
+//			rewardBalanceChanged |= accountNum == STAKING_FUNDING_ACCOUNT_NUMBER;
+			checksFundingBalanceChange(changes, accountNum);
+
+			// Update BALANCE and STAKE_PERIOD_START in the pending changes for this account, if reward-eligible
+			if (isRewardable(account, changes, latestEligibleStart)) {
+				payReward(account, changes, accountNum);
+			}
+			// Update any STAKED_TO_ME side effects of this change
+			n = updateStakedToMeSideEffects(i, pendingChanges, latestEligibleStart);
+		}
+	}
+
+	private void checksFundingBalanceChange(final Map<AccountProperty, Object> changes, final long accountNum) {
+		if (changes.containsKey(AccountProperty.BALANCE)) {
+			final long newBalance = (long) changes.get(AccountProperty.BALANCE);
+			if (accountNum == STAKING_FUNDING_ACCOUNT_NUMBER) {
+				rewardBalanceChanged = true;
+				newRewardBalance = newBalance;
+			}
+		}
+	}
+
+
+	private void updateFundingRewardBalances(
+			final EntityChangeSet<AccountID, MerkleAccount, AccountProperty> pendingChanges) {
+		final var rewardsPaid = rewardCalculator.rewardsPaidInThisTxn();
+		rewardBalanceChanged |= rewardsPaid > 0;
+		if (rewardsPaid > 0) {
+			final var rewardAccountI = stakeChangeManager.findOrAdd(800L, pendingChanges);
+			newRewardBalance += stakeChangeManager.updateBalance(-rewardsPaid, rewardAccountI, pendingChanges);
+		}
+	}
+
+	private void updateNodeStakes(final EntityChangeSet<AccountID, MerkleAccount, AccountProperty> pendingChanges) {
+		for (int i = 0, n = pendingChanges.size(); i < n; i++) {
+			final var account = pendingChanges.entity(i);
+			final var changes = pendingChanges.changes(i);
+
+			final var curNodeId = (account != null) ? account.getStakedId() : 0L;
+			final var newNodeId = stakeChangeManager.getNodeStakeeNum(changes);
+			if (curNodeId < 0 && curNodeId != newNodeId) {
+				// Node stakee has been replaced, withdraw stakeRewarded or stakeNotRewarded from ex-stakee based on
+				// isDeclineReward option
+				stakeChangeManager.withdrawStake(
+						Math.abs(curNodeId),
+						account.getBalance() + account.getStakedToMe(),
+						stakeChangeManager.finalDeclineRewardGiven(account, changes));
+			}
+			if (newNodeId < 0) {
+				// Award updated stake to new node stakee to the fields stakeRewarded or stakeNotRewarded from
+				// ex-stakee based on isDeclineReward option
+				stakeChangeManager.awardStake(
+						Math.abs(newNodeId),
```

Comment:
Shouldn't this be `Math.abs(newNodeId + 1)`  ?

Reply:
Yes.. it should be. Thanks @anighanta